src/query/query_executor.c

```diff
@@ -7919,11 +8016,10 @@ qexec_intprt_fnc (THREAD_ENTRY * thread_p, XASL_NODE * xasl, XASL_STATE * xasl_s
 
 	      if (cte_start_new_iteration)
 		{
-		  recursive_iterations++;
-		  if (recursive_iterations >= xasl->max_iterations)
+		  if (++recursive_iterations >= xasl->max_iterations)
 		    {
-		      max_recursive_iterations_reached = true;
-		      break;
+		      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_CTE_MAX_RECURSION_REACHED, 1, xasl->max_iterations);
+		      return S_ERROR;
```

Comment:
is it okay that return S_ERROR instead of break?

Reply:
When a break is encountered in the current code, er_set() is executed at line 8190, and eventually “return S_ERROR;” at line 8205 is encountered.